File: android/src/main/kotlin/io/github/kokemus/cross_bluetooth_api/events/GattServerDisconnectedEvent.kt
package io.github.kokemus.cross_bluetooth_api.events

class GattServerDisconnectedEvent(private val deviceId: String) :
    Event(EventName.GATT_SERVER_DISCONNECTED) {
    override fun toMap(): Map<String, Any> {
        return super.toMap() + mapOf("deviceId" to deviceId)
    }
}

File: ios/cross_bluetooth_api/Sources/cross_bluetooth_api/Events/GattServerDisconnectedEvent.swift
import Foundation
import Flutter

public class GattServerDisconnectedEvent: Event {
    let deviceId: String
    
    init(deviceId: String) {
        self.deviceId = deviceId
        super.init(name: .gattserverdisconnected)
    }
    
    override func toMap() -> [String:AnyObject] {
        var map = super.toMap()
        map["deviceId"] = deviceId as AnyObject
        return map
    }
}